jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/jdt/cdi/ManagedBeanDiagnosticsCollector.java

```diff
@@ -202,41 +202,62 @@ else if (!hasParameterizedInjectConstructor)
+                     * 
+                     * see: 
+                     * https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#declaring_producer_method
+                     * 
+                     * note:
+                     * we need to check for bean defining annotations first to make sure the managed bean is discovered.
+                     * 
+                     */
+                    invalidParamsCheck(unit, diagnostics, type, ManagedBeanConstants.PRODUCES); 
+                }
+            }
+
+        } catch (JavaModelException e) {
+            Activator.logException("Cannot calculate diagnostics", e);
+        }
+    }
+    
+    
+    private void invalidParamsCheck(ICompilationUnit unit, List<Diagnostic> diagnostics, IType type, String name) throws JavaModelException {
+        for (IMethod method : type.getMethods()) {
+            IAnnotation Annotation = null;
 
-                    if (InjectAnnotation == null)
-                        continue;
+            for (IAnnotation annotation : method.getAnnotations()) {
+                if (annotation.getElementName().equals(name))
+                    Annotation = annotation;
+            }
 
-                    Set<String> invalidInjectAnnotations = new TreeSet<>();
-                    for (ILocalVariable param : method.getParameters()) {
-                        for (IAnnotation annotation : param.getAnnotations()) {
-                            if (ManagedBeanConstants.INVALID_INJECT_PARAMS.contains(annotation.getElementName())) {
-                                invalidInjectAnnotations.add("@" + annotation.getElementName());
-                            }
-                        }
-                    }
+            if (Annotation == null)
+                continue;
 
-                    if(!invalidInjectAnnotations.isEmpty()) {
-                        String label = createInvalidInjectLabel(invalidInjectAnnotations);
-                        diagnostics.add(createDiagnostic(method, unit, label, ManagedBeanConstants.DIAGNOSTIC_CODE_INVALID_INJECT_PARAM));
+            Set<String> invalidAnnotations = new TreeSet<>();
+            for (ILocalVariable param : method.getParameters()) {
+                for (IAnnotation annotation : param.getAnnotations()) {
+                    if (ManagedBeanConstants.INVALID_INJECT_PARAMS.contains(annotation.getElementName())) {
+                        invalidAnnotations.add("@" + annotation.getElementName());
                     }
-
                 }
             }
 
-        } catch (JavaModelException e) {
-            Activator.logException("Cannot calculate diagnostics", e);
+            if(!invalidAnnotations.isEmpty()) {
+                String label = createInvalidInjectLabel(name, invalidAnnotations);
+                diagnostics.add(createDiagnostic(method, unit, label, ManagedBeanConstants.DIAGNOSTIC_CODE_INVALID_INJECT_PARAM));
```

Comment:
Should this use DIAGNOSTIC_CODE_INVALID_INJECT_PARAM or DIAGNOSTIC_CODE_INVALID_PRODUCES_PARAM depending on the scenario?